### PR TITLE
Fix issue when build upstream kernel

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/operating_system.py [hunk 1]
```diff
@@ -282,15 +282,20 @@ def install_packages(
...
+    def package_installed(self, package: Union[str, Tool, Type[Tool]]) -> bool:
         """
         Query if a package/tool is installed on the node.
         Return Value - bool
         """
         package_name = self.__resolve_package_name(package)
-        return self._package_exists(package_name)
+        return self._package_installed(package_name)
+
+    def package_existed(self, package: str) -> bool:
```
Comment (squirrelsc) [inline lisa/operating_system.py]: `is_package_in_repo`

File: lisa/operating_system.py [hunk 2]
```diff
@@ -282,15 +282,20 @@ def install_packages(
         package_names = self._get_package_list(packages)
         self._install_packages(package_names, signed)
 
-    def package_exists(
-        self, package: Union[str, Tool, Type[Tool]], signed: bool = True
-    ) -> bool:
+    def package_installed(self, package: Union[str, Tool, Type[Tool]]) -> bool:
```
Comment (squirrelsc) [inline lisa/operating_system.py]: It doesn't need to change the name, if the next one is renamed.

File: lisa/operating_system.py [hunk 3]
```diff
@@ -282,15 +282,20 @@ def install_packages(
...
         package_name = self.__resolve_package_name(package)
-        return self._package_exists(package_name)
+        return self._package_installed(package_name)
+
+    def package_existed(self, package: str) -> bool:
+        """
+        Query if a package/tool exists in the repo
+        Return Value - bool
+        """
+        return self._package_existed(package)
```
Comment (squirrelsc) [inline lisa/operating_system.py]: Does it need to call `package_name = self.__resolve_package_name(package)`?

File: lisa/operating_system.py [hunk 4]
```diff
@@ -578,6 +586,42 @@ def _package_exists(self, package: str, signed: bool = True) -> bool:
             return True
         return False
 
+    def _package_existed(self, package: str) -> bool:
+        command = f"apt-cache policy {package}"
+        result = self._node.execute(command, sudo=True, shell=True)
+        package_pattern = re.compile(r"Candidate: ((?!none)).*", re.M)
```
Comment (squirrelsc) [inline lisa/operating_system.py]: move it to class level.

File: lisa/operating_system.py (hunk 2 above)
Comment (LiliDeng) [reply lisa/operating_system.py]: package_exists is not clear, so I want to rename it

File: lisa/operating_system.py (hunk 3 above)
Comment (LiliDeng) [reply lisa/operating_system.py]: to be frank, I am not clear what is the usage of __resolve_package_name

Comment (squirrelsc) [reply lisa/operating_system.py]: Some packages are passed in as Tool type, not a string. The method will convert it to a package str by calling the `package_name` method.

File: lisa/operating_system.py (hunk 2 above)
Comment (squirrelsc) [reply lisa/operating_system.py]: it's a method on os, so it means if the package exists in the current os.

File: lisa/operating_system.py (hunk 1 above)
Comment (LiliDeng) [reply lisa/operating_system.py]: updated

File: lisa/operating_system.py (hunk 2 above)
Comment (LiliDeng) [reply lisa/operating_system.py]: revert

File: lisa/operating_system.py (hunk 3 above)
Comment (LiliDeng) [reply lisa/operating_system.py]: updated

File: lisa/operating_system.py (hunk 4 above)
Comment (LiliDeng) [reply lisa/operating_system.py]: moved

File: lisa/tools/xfstests.py [hunk 5]
```diff
@@ -130,7 +130,8 @@ def _install_dep(self) -> None:
         # be installed if one of packages is not available in that distro,
         # so here install it one by one.
         for package in list(package_list):
-            posix_os.install_packages(package)
+            if posix_os.is_package_in_repo(package):
```
Comment (squirrelsc) [inline lisa/tools/xfstests.py:136]: What's the else? raise an exception?

Comment (squirrelsc) [reply lisa/tools/xfstests.py:136]: If not to raise an exception, add comments in code to explain it.

Comment (LiliDeng) [reply lisa/tools/xfstests.py:136]: added.

File: lisa/tools/xfstests.py [hunk 6]
```diff
@@ -128,9 +128,13 @@ def _install_dep(self) -> None:
...
         # be installed if one of packages is not available in that distro,
-        # so here install it one by one.
+        # so here install it one by one
         for package in list(package_list):
-            posix_os.install_packages(package)
+            # to make code simple, put all packages needed by one distro in one list.
+            # the package name may be different for the different sku of the
+            #  same distro. so, install it when the package exists in the repo.
+            if posix_os.is_package_in_repo(package):
+                posix_os.install_packages(package)
```
Comment (squirrelsc) [inline lisa/tools/xfstests.py:137]: Another way is to add a boolean on `install_packages` called `ignore_non_exists`. It makes the logic simpler, and easy to read.

**Review by squirrelsc — APPROVED** [review]

File: lisa/tools/xfstests.py (hunk 6 above)
Comment (LiliDeng) [reply lisa/tools/xfstests.py:137]: I will raise another PR for it, use #1641 to track.

**Pull request merged**